### PR TITLE
updated rubocop version to latest

### PR DESCRIPTION
**Pull request by md-islam-rewind**
## Description of change
Updated the rubocop version to 1.10 (from 0.89). We are enabling most of the new cops (rules) that came with the new version. There are few disabled (see in the config file).

### CODE REVIEW

File: rewind-ruby-style.gemspec [hunk 1]
```diff
@@ -20,5 +20,5 @@ Gem::Specification.new do |s|
     'source_code_uri' => "https://github.com/rewindio/ruby-style-configs/tree/v#{s.version}"
   }
 
-  s.add_dependency('rubocop', '>= 0.85', '< 0.89')
+  s.add_dependency('rubocop', '~1.10')
```
Comment (jhuff078) [inline rewind-ruby-style.gemspec:23]: `~>`?

**Review by jhuff078 — APPROVED** [review]

File: rewind-ruby-style.gemspec (hunk 1 above)
Comment (cschlack) [reply rewind-ruby-style.gemspec:23]: That, or `>= 1.10`

File: rubocop.yml [hunk 2]
```diff
@@ -24,16 +25,6 @@ Metrics/ClassLength:
 Metrics/CyclomaticComplexity:
   Enabled: false
```
Comment (md-islam-rewind) [inline rubocop.yml:27]: Change between old line 27:81 (number on the left) are removal of redundant rules (rubocop doesn't require us to explicitly enable them any more- enabled by default)

**Review by sanket0354 — APPROVED** [review]

File: rewind-ruby-style.gemspec [hunk 3]
```diff
@@ -16,9 +16,11 @@ Gem::Specification.new do |s|
...
   s.files = ['rubocop.yml', 'reek.yml']
 
+  s.required_ruby_version = '>= 2.6'
+
   s.metadata = {
     'source_code_uri' => "https://github.com/rewindio/ruby-style-configs/tree/v#{s.version}"
   }
 
-  s.add_dependency('rubocop', '>= 0.85', '< 0.89')
+  s.add_dependency('rubocop', '~> 1.10')
```
Comment (dnorth98) [inline rewind-ruby-style.gemspec:25]: So before you merge this, you will have to go update ALL of the github actions in all repos to install this version of Rubocop or they will all fail I believe

For example

https://github.com/rewindio/rewind-app/blob/44da5161dff59704f2a798c85c6acb7a26ce62f7/.github/workflows/style.yml#L33

That style workflow exists in all repos and needs to be updated in them all as it's what installs Rubocop

**Review by dnorth98 — APPROVED** [review]

File: rewind-ruby-style.gemspec (hunk 3 above)
Comment (cschlack) [reply rewind-ruby-style.gemspec:25]: @dnorth98: We're going to change the workflows to use the rubocop version (and other dependencies) from the Gemfile.lock

The plan is to be able to decide for each repository separately when to update this gem, so we won't have to do all at once.

File: rubocop.yml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 AllCops:
   DisplayCopNames: true
   TargetRubyVersion: 2.6
+  SuggestExtensions: false
```
Comment (cschlack) [inline rubocop.yml:4]: I would recommend disabling **NewCops** and only enable them after review and within a planned release of this gem. This way we're more flexible updating rubocop without breaking .workflows and such

```suggestion
  SuggestExtensions: false
  NewCops: disable
```

**Review by cschlack — APPROVED** [review]

**Comment by gregfletch** [comment]
Couple of comments if this PR is still active. Rubocop version should be updated to new current version, 1.14.0. Also, I would recommend adding other rubocop modules, specifically rubocop-rails (current version: 2.10.1), rubocop-performance (current version: 1.11.3), rubocop-rspec (current version: 2.3.0) (or rubocop-minitest (current version: 0.12.1), depending on which is used internally). Note, you will have to update the `.rubocop.yml` file and add the `require` block at the top of the file.

**Pull request closed without merging**